WordPress/Classes/ViewRelated/Post/Categories/PostCategoriesViewController.swift

```diff
@@ -0,0 +1,281 @@
+
+    @objc init(blog: Blog, currentSelection: [PostCategory]?, selectionMode: CategoriesSelectionMode) {
+        self.blog = blog
+        self.selectionMode = selectionMode
+        self.originalSelection = currentSelection
+        super.init(style: .grouped)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureTableView()
+        configureView()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        reloadCategories()
+        if !hasSyncedCategories {
+            syncCategories()
+        }
+    }
+
+    private func configureTableView() {
+        tableView.accessibilityIdentifier = "CategoriesList"
+        tableView.cellLayoutMarginsFollowReadableWidth = true
+        tableView.register(WPTableViewCell.self, forCellReuseIdentifier: Constants.categoryCellIdentifier)
+        WPStyleGuide.configureAutomaticHeightRows(for: tableView)
+    }
+
+    private func configureView() {
+        WPStyleGuide.configureColors(view: view, tableView: tableView)
+        refreshControl = UIRefreshControl()
+        refreshControl!.addTarget(self, action: #selector(refreshCategoriesWithInteraction), for: .valueChanged)
+        let rightBarButtonItem = UIBarButtonItem(image: UIImage(named: "icon-post-add"), style: .plain, target: self, action: #selector(showAddNewCategory))
+        switch selectionMode {
+        case .post:
+            navigationItem.rightBarButtonItem = rightBarButtonItem
+            title = NSLocalizedString("Post Categories", comment: "Title for selecting categories for a post")
+        case .parent:
+            navigationItem.rightBarButtonItem = rightBarButtonItem
+            title = NSLocalizedString("Parent Category", comment: "Title for selecting parent category of a category")
+        case .blogDefault:
+            title = NSLocalizedString("Default Category", comment: "Title for selecting a default category for a posty")
+        }
+    }
+
+    @objc private func refreshCategoriesWithInteraction() {
+        syncCategories()
+    }
+
+    @objc private func showAddNewCategory() {
+        let addCategoriesViewController = WPAddPostCategoryViewController(blog: blog)
+        addCategoriesViewController!.delegate = self
+        let addCategoriesNavigationControlloer = UINavigationController(rootViewController: addCategoriesViewController!)
+        navigationController?.modalPresentationStyle = .formSheet
+        present(addCategoriesNavigationControlloer, animated: true, completion: nil)
+    }
+
+    private func syncCategories() {
+        guard let context = blog.managedObjectContext else {
+            return
+        }
+        let service  = PostCategoryService(managedObjectContext: context)
+        service.syncCategories(for: blog, success: { [weak self] in
+            self?.reloadCategories()
```

Comment:
I actually prefer to leave self as weak here instead of adding the guard for both blocks. I also prefer those to be inline. Unless you have a strong preference